airbyte-webapp/src/components/ui/CheckBox/CheckBox.tsx

```diff
@@ -1,49 +1,40 @@
 import { faCheck, faMinus } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import React from "react";
-import styled from "styled-components";
+import classNames from "classnames";
+import React, { InputHTMLAttributes } from "react";
 
-const CheckBoxInput = styled.input`
-  opacity: 0;
-  width: 0;
-  height: 0;
-  margin: 0;
-  position: absolute;
-`;
+import styles from "./CheckBox.module.scss";
 
-const CheckBoxContainer = styled.label<{
-  checked?: boolean;
+type CheckBoxSize = "lg" | "sm";
+
+export interface CheckBoxProps extends InputHTMLAttributes<HTMLInputElement> {
   indeterminate?: boolean;
-}>`
-  height: 18px;
-  min-width: 18px;
-  border: 1px solid
-    ${({ theme, checked, indeterminate }) => (checked || indeterminate ? theme.primaryColor : theme.greyColor20)};
-  background: ${({ theme, checked, indeterminate }) =>
-    checked || indeterminate ? theme.primaryColor : theme.whiteColor};
-  color: ${({ theme }) => theme.whiteColor};
-  text-align: center;
-  border-radius: 4px;
-  font-size: 13px;
-  line-height: 13px;
-  display: inline-block;
-  padding: 1px 0;
-  cursor: pointer;
-  vertical-align: top;
-  position: relative;
-`;
+  elSize?: CheckBoxSize;
```

Comment:
Nit:
```suggestion
  checkboxSize?: CheckBoxSize;
```

Reply:
Could we potentially go with `size` here just? It feels given this is the checkbox component whose size we control, we should not try to prefix it with any other prefix, but just use `size` plainly. This would help us also getting to a bit more aligned property names across components.

Reply:
That was the first I've tried 😄 But unfortunately the native `<input />` has its own `size` property.
We have two options:
1. Omit the native `size` prop and use our one for sizing:
`export interface CheckBoxProps extends Omit<InputHTMLAttributes<HTMLInputElement>, "size"> {
  indeterminate?: boolean;
  size?: CheckBoxSize;
}`
3. Use another prop name for sizing(elSize, checkboxSize, etc.)

@timroes IMO the first option doesn't look dangerous 😬 WDYT?

Reply:
Given that the `size` attribute on an input type="checkbox" has anyway no effect (since it's only for text, search, tel, url, email, and password), I think we could safely omit it, so we can nicely reuse it.

Reply:
Great! Will omit it 👍

Reply:
Did a try to omit the `size` prop, and here is what I've discovered:
On some higher levels, we through the props via `{...props}`
Here is an example:
<img width="1029" alt="image" src="https://user-images.githubusercontent.com/20929344/207401168-ca593742-1dd0-4f23-bf26-a43bb2f6dbc5.png">

And since we try to change the internal `size` prop, it argues that.

Options we have:
1. Make unavailable to pass props like this way - `{...props}`
2. Use `checkboxSize` as a prop name for sizing

@timroes

Reply:
Don't wanna block the PR because of the prop name, will change it to `checkboxSize`